team_access_test.go

```diff
@@ -107,7 +136,23 @@ func TestTeamAccessesAdd(t *testing.T) {
 		}
 	})
 
+	t.Run("with invalid custom options", func(t *testing.T) {
+		options := TeamAccessAddOptions{
+			Access:    Access(AccessRead),
+			Runs:      RunsPermission(RunsPermissionApply),
+			Team:      tmTest,
+			Workspace: wTest,
+		}
+
+		_, err := client.TeamAccess.Add(ctx, options)
+
+		assert.EqualError(t, err, "invalid attribute\n\nRuns is read-only when access level is 'read'; use the 'custom' access level to set this attribute.")
+	})
```

Comment:
nit: i've leaned away from testing for exact error that comes from the API, thinking that the responsibility for testing the correct error occurs should be on the API side. i was also worried that it would make these tests too brittle. i've been testing that an error occurred instead, like this:

```suggestion
                assert.Error(t, err)
```

the only error messages i test for exactly are the error messages we have added to go-tfe directly.

Reply:
Biggest problem with that is that you can't just assert a general error without introducing a potential false positive. _Any_ error would make this test pass, in other words. 

Still though, brittle point taken and I'm happy to just change it here for now!